### PR TITLE
TAJO-1971: Replace 'for' loop with 'foreach'

### PR DESCRIPTION
**Pull request by dkhwangbo**
Reports for loops which iterate over collections or arrays, and can be replaced with the foreach iteration syntax, available in Java 5 and newer, to get the improvement of readability and so on.

### CODE REVIEW

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java [hunk 1]
```diff
@@ -241,8 +241,8 @@ private void mergeLocalToBuilder() {
 			builder.setId(this.id.getProto());
 		}
 		if (fragments != null) {
-			for (int i = 0; i < fragments.size(); i++) {
-				builder.addFragments(fragments.get(i));
+			for (FragmentProto fragment : fragments) {
+				builder.addFragments(fragment);
```
Comment (eminency) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: Use 2 spaces instead of tab character.

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java [hunk 2]
```diff
@@ -258,9 +258,9 @@ private void mergeLocalToBuilder() {
 		  builder.setInterQuery(this.interQuery);
 		}
     if (this.fetches != null) {
-      for (int i = 0; i < fetches.size(); i++) {
-        builder.addFetches(fetches.get(i).getProto());
-      }
+		for (FetchImpl fetche : fetches) {
+			builder.addFetches(fetche.getProto());
+		}
```
Comment (eminency) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: Use 2 spaces instead of tab character.

**Comment by hyunsik** [comment]
The patch looks good to me. If you fix the coding convention eminency mentioned, I'll finish the review.

**Comment by dkhwangbo** [comment]
@eminency @hyunsik 
Thanks for your reply. I fix some code line with TAJO coding convention.

File: tajo-catalog/tajo-catalog-drivers/tajo-hive/src/test/java/org/apache/tajo/catalog/store/TestHiveCatalogStore.java [hunk 3]
```diff
@@ -416,8 +416,8 @@ private void testAddPartition(URI uri, String tableName, String partitionName) t
 
     List<PartitionKeyProto> partitionKeyList = new ArrayList<>();
     String[] partitionNames = partitionName.split("/");
-    for(int i = 0; i < partitionNames.length; i++) {
-      String[] eachPartitionName = partitionNames[i].split("=");
+    for (String partitionName1 : partitionNames) {
```
Comment (hyunsik) [inline tajo-catalog/tajo-catalog-drivers/tajo-hive/src/test/java/org/apache/tajo/catalog/store/TestHiveCatalogStore.java]: How about just ``partitionName``?

File: tajo-catalog/tajo-catalog-drivers/tajo-hive/src/test/java/org/apache/tajo/catalog/store/TestHiveCatalogStore.java [hunk 4]
```diff
@@ -454,8 +454,8 @@ private void testAddPartitions(URI uri, String tableName, List<String> partition
 
       List<PartitionKeyProto> partitionKeyList = new ArrayList<>();
       String[] split = partitionName.split("/");
```
Comment (hyunsik) [inline tajo-catalog/tajo-catalog-drivers/tajo-hive/src/test/java/org/apache/tajo/catalog/store/TestHiveCatalogStore.java]: It should be splits.

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java [hunk 5]
```diff
@@ -35,113 +35,113 @@
...
-			String outputTable, boolean clusteredOutput,
-			PlanProto.LogicalNodeTree plan, QueryContext queryContext,
-									DataChannel dataChannel, Enforcer enforcer, String queryMasterHostAndPort) {
-		this.id = id;
-		this.fragments = fragments;
-		this.outputTable = outputTable;
-		this.clusteredOutput = clusteredOutput;
-		this.plan = plan;
-		this.isUpdated = true;
+  private TaskAttemptId id;
```
Comment (hyunsik) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: It is right fix. But, it should be resolved in another jira. Could you create new jira for it?

**Comment by hyunsik** [comment]
The patch looks good to me. Could you change some suffix ``1`` in partitionName1? If you have concern with similar look between partitionName and partitionNames, it would be better if you use ``aName`` instead of partitionName.

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java (hunk 5 above)
Comment (dkhwangbo) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: You mean, rollback this and fix with another jira? got it.

File: tajo-catalog/tajo-catalog-drivers/tajo-hive/src/test/java/org/apache/tajo/catalog/store/TestHiveCatalogStore.java (hunk 3 above)
Comment (dkhwangbo) [inline tajo-catalog/tajo-catalog-drivers/tajo-hive/src/test/java/org/apache/tajo/catalog/store/TestHiveCatalogStore.java]: ```partitionName``` is already in use as parameter of ```testAddPartition()``` . So, I'll rename ```partitionName1``` to ```Name``` as you mentioned.

**Comment by dkhwangbo** [comment]
@hyunsik Thanks for your review and detailed comment! I apply your comment.

File: tajo-common/src/test/java/org/apache/tajo/datum/TestIntervalDatum.java [hunk 6]
```diff
@@ -122,8 +122,8 @@ public final void testOperation() throws Exception {
 
     // date '2001-10-01' - integer '7' ==>	date '2001-09-24'
     datum = DatumFactory.createDate(2001, 10, 1);
-    for (int i = 0; i < datums.length; i++) {
-      Datum result2 = datum.minus(datums[i]);
+    for (Datum datum1 : datums) {
```
Comment (jihoonson) [inline tajo-common/src/test/java/org/apache/tajo/datum/TestIntervalDatum.java]: Well, this is valid but it looks not a good convention. How about to change ```eachDatum```?

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java [hunk 7]
```diff
@@ -35,113 +35,113 @@
...
-		this.clusteredOutput = clusteredOutput;
-		this.plan = plan;
-		this.isUpdated = true;
-    this.queryContext = queryContext;
-    this.queryContext = queryContext;
-    this.dataChannel = dataChannel;
-    this.enforcer = enforcer;
-		this.queryMasterHostAndPort = queryMasterHostAndPort;
-	}
+        private TaskAttemptId id;
```
Comment (jihoonson) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: Please remove unnecessary change.

File: tajo-thirdparty/asm/src/main/java/org/apache/tajo/org/objectweb/asm/commons/AnalyzerAdapter.java [hunk 8]
```diff
@@ -177,33 +177,32 @@ protected AnalyzerAdapter(final int api, final String owner,
...
-                locals.add(Opcodes.DOUBLE);
-                locals.add(Opcodes.TOP);
-                break;
-            case Type.ARRAY:
-                locals.add(types[i].getDescriptor());
-                break;
-            // case Type.OBJECT:
-            default:
-                locals.add(types[i].getInternalName());
+                case Type.BOOLEAN:
```
Comment (jihoonson) [inline tajo-thirdparty/asm/src/main/java/org/apache/tajo/org/objectweb/asm/commons/AnalyzerAdapter.java]: Please remove unnecessary change.

**Comment by jihoonson** [comment]
Overall, this patch looks good to me. Please consider my comments.

**Comment by dkhwangbo** [comment]
@jihoonson Thanks for your review! I apply your comments.

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java [hunk 9]
```diff
@@ -258,9 +258,9 @@ private void mergeLocalToBuilder() {
 		  builder.setInterQuery(this.interQuery);
 		}
     if (this.fetches != null) {
-      for (int i = 0; i < fetches.size(); i++) {
-        builder.addFetches(fetches.get(i).getProto());
-      }
+			for (FetchImpl fetche : fetches) {
```
Comment (jihoonson) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: Same for this change. Please use 2 spaces instead of tab.

Comment (dkhwangbo) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: Oh, I really sorry about it. I missed. Change it right now.

File: tajo-catalog/tajo-catalog-server/src/test/java/org/apache/tajo/catalog/CatalogTestingUtil.java [hunk 10]
```diff
@@ -144,14 +144,14 @@ public static PartitionDesc buildPartitionDesc(String partitionName) {
     String[] partitionNames = partitionName.split("/");
 
     List<PartitionKeyProto> partitionKeyList = new ArrayList<>();
-    for(int i = 0; i < partitionNames.length; i++) {
-      String [] splits = partitionNames[i].split("=");
+    for (String partitionName1 : partitionNames) {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-server/src/test/java/org/apache/tajo/catalog/CatalogTestingUtil.java]: This is also valid, but it looks not a good convention

File: tajo-core/src/main/java/org/apache/tajo/engine/planner/physical/DistinctGroupbyHashAggregationExec.java [hunk 11]
```diff
@@ -366,8 +365,8 @@ public HashAggregator(GroupbyNode groupbyNode, Schema schema) throws IOException
 
       Column[] keyColumns = groupbyNode.getGroupingColumns();
       Column col;
-      for (int idx = 0; idx < keyColumns.length; idx++) {
-        col = keyColumns[idx];
+      for (Column keyColumn : keyColumns) {
+        col = keyColumn;
```
Comment (jihoonson) [inline tajo-core/src/main/java/org/apache/tajo/engine/planner/physical/DistinctGroupbyHashAggregationExec.java]: ```col``` is redundant.

File: tajo-thirdparty/asm/src/main/java/org/apache/tajo/org/objectweb/asm/commons/AdviceAdapter.java [hunk 12]
```diff
@@ -419,9 +419,9 @@ public void visitMethodInsn(final int opcode, final String owner,
         mv.visitMethodInsn(opcode, owner, name, desc);
         if (constructor) {
             Type[] types = Type.getArgumentTypes(desc);
-            for (int i = 0; i < types.length; i++) {
+            for (Type type1 : types) {
```
Comment (jihoonson) [inline tajo-thirdparty/asm/src/main/java/org/apache/tajo/org/objectweb/asm/commons/AdviceAdapter.java]: This is valid but it looks not a good convention.

File: tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java [hunk 13]
```diff
@@ -259,8 +256,8 @@ private void mergeLocalToBuilder() {
 		  builder.setInterQuery(this.interQuery);
 		}
     if (this.fetches != null) {
-      for (int i = 0; i < fetches.size(); i++) {
-        builder.addFetches(fetches.get(i));
+      for (FetchProto fetche : fetches) {
```
Comment (jihoonson) [inline tajo-core/src/main/java/org/apache/tajo/engine/query/TaskRequestImpl.java]: ```fetche``` looks a typo.

**Comment by dkhwangbo** [comment]
@jihoonson Hi! Thanks for your comment. I apply your mention. Please check it again.

**Comment by jihoonson** [comment]
+1 LGTM
